CHANGELOG.md

```diff
@@ -1,3 +1,19 @@
+## 3.16.0 (June 7, 2023)
+FEATURES:
+* Add support for LDAP secrets engine: ([#1859](https://github.com/hashicorp/terraform-provider-vault/pull/1859))
+* Add new data source `vault_auth_backends`: ([#1827](https://github.com/hashicorp/terraform-provider-vault/pull/1827))
+* Support allowed_domains_template on ssh_secret_backend_role. Fixes hashicorp#1675: ([#1676](https://github.com/hashicorp/terraform-provider-vault/pull/1676))
+
+IMPROVEMENTS:
+* Add support for retrying kv-v2 secret data writes: ([#1887](https://github.com/hashicorp/terraform-provider-vault/pull/1887))
+* Add back support for deriving the provider namespace from the Vault token's: ([#1841](https://github.com/hashicorp/terraform-provider-vault/pull/1841))
+
+BUGS:
+* check version for allowed_domains_template field: ([#1881](https://github.com/hashicorp/terraform-provider-vault/pull/1881))
+* Fix DB engine password overwrite: ([#1876](https://github.com/hashicorp/terraform-provider-vault/pull/1876))
+* secrets/ldap: fix bindpass getting overwritten: ([#1875](https://github.com/hashicorp/terraform-provider-vault/pull/1875))
```

Comment:
I wonder if @vinay-gopalan 's comment above applies for the LDAP bug fix here as well, since it was never officially released.